### PR TITLE
Fix/optimizer

### PR DESCRIPTION
**Pull request by SunCreation**
It was incorrect to define only one optimizer and transfer it each processes.
It should be defined at each processes and the below code is to merge whole gradients in training step.
```python3
mx.optimizer_step(optimizer)
```

### CODE REVIEW

**Review by YejinHwang909 — APPROVED** [review]
You're the best. I'm proud of you! 😆

**Comment by SunCreation** [comment]
That's what I think of you:)
